app/controllers/application_controller.rb

```diff
@@ -5,6 +5,158 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+
+    enable :sessions
+    set :session_secret, "secret_fwitter"
+  end
+
+  get '/' do
+    erb :index
+  end
+
+  get '/login' do
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :'users/login'
+    end
+  end
+
+  post '/login' do
+    user = User.find_by(:username => params[:username])
+
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      redirect '/tweets'
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    elsif !logged_in?
+      erb :'users/create_user'
+    end
+  end
+
+  post '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    end
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.